include/nbl/builtin/shader/loader/gltf/fragment_impl.glsl

```diff
@@ -0,0 +1,96 @@
+#include "nbl/builtin/shader/loader/gltf/common.glsl"
+
+#ifndef _NBL_FRAG_INPUTS_DEFINED_
+#define _NBL_FRAG_INPUTS_DEFINED_
+layout (location = _NBL_F_IN_LOCAL_POSITION_ATTRIBUTE_ID) in vec3 LocalPos;
+layout (location = _NBL_F_IN_VIEW_POS_ATTRIBUTE_ID) in vec3 ViewPos;
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _NBL_F_IN_UV_ATTRIBUTE_ID) in vec2 UV_0;
+#endif
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout (location = _NBL_F_IN_COLOR_ATTRIBUTE_ID) in vec4 Color_0;
+#endif // _DISABLE_COLOR_ATTRIBUTES
+#endif //_NBL_FRAG_INPUTS_DEFINED_
+
+#ifndef _NBL_FRAG_OUTPUTS_DEFINED_
+#define _NBL_FRAG_OUTPUTS_DEFINED_
+layout (location = 0) out vec4 OutColor;
+#endif //_NBL_FRAG_OUTPUTS_DEFINED_
+
+layout( push_constant, row_major ) uniform Block {
+	vec4 baseColorFactor;
+	float metallicFactor;
+	float roughnessFactor;
+	
+	vec3 emissiveFactor;
+	uint alphaMode;
+	float alphaCutoff;
```

Comment:
do any of the factors have constraints like needing to be positive, or in the [0,1] range?

`alphaCutoff` probably has such a constraint